Additional install script option to install nightly/release builds

carlos-logro opened the pull request:
issue: #219 

# Changes
Asks user which version he would like installed and from there do the necessary installation process.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

carlos-logro commented:
/uncc @steveodonovan 
/cc @a-roberts @dibbles

a-roberts commented on webhooks-extension/test/install_dashboard_and_extension.sh:
```diff
@@ -16,7 +16,37 @@ export test_dir="${tekton_repo_dir}/webhooks-extension/test"
-echo "Installing Dashoard and webhooks extension into $DASHBOARD_INSTALL_NS namespace"
-install_webhooks_extension $DASHBOARD_INSTALL_NS
-install_dashboard $DASHBOARD_INSTALL_NS
+# Ask user what verison he will like to install
```
Unnecessary comment given the below echo

a-roberts commented on webhooks-extension/test/util.sh:
```diff
@@ -126,69 +126,105 @@ function install_tekton() {
+      # Notify and depending on [version] we will ko apply or kubectl apply gcr yaml
+      echo -e "\n\nDashboard source detected at $dashboard_dir"
+    else
+      # If not, notify and ask if he wants to clone it
```
Don't need this comment

a-roberts commented on webhooks-extension/test/util.sh:
```diff
@@ -126,69 +126,105 @@ function install_tekton() {
+        read -p "Do you wish to git clone and ko apply the master version? " -n 1 -r cloneAndApply
+      fi
+      if [[ $cloneAndApply =~ ^[Yy]$ ]] || [[ $version == "3" ]]; then
+        # Dashboard source cloned
```
Don't need

a-roberts commented on webhooks-extension/test/util.sh:
```diff
@@ -126,69 +126,105 @@ function install_tekton() {
+    if [[ $version == "3" ]]; then
+      pushd $GOPATH/src/github.com/tektoncd/dashboard
+      docker login
+      npm install
```
Should be `npm ci`? You use it later so wondering if npm install is intentional here

carlos-logro replied:
definitely, thanks Adam!

carlos-logro commented:
/retest

a-roberts commented:
@carlos-logro would be useful to include some docs for this PR too in order to advocate the script's usage, what do you think?

carlos-logro commented:
@a-roberts yes it will indeed help user out! ill go ahead and added it

carlos-logro commented:
@a-roberts Updated the doc! let me know if the wording is on point

a-roberts commented on webhooks-extension/test/install_dashboard_and_extension.sh:
```diff
@@ -16,7 +16,36 @@ export test_dir="${tekton_repo_dir}/webhooks-extension/test"
+  VERSION_TEXT="DEVELOPMENT VERSIONS"
+fi
+
+echo
```
What's this doing here?

carlos-logro replied:
thought I removed this echo....

a-roberts commented on webhooks-extension/test/install_dashboard_and_extension.sh:
```diff
@@ -16,7 +16,36 @@ export test_dir="${tekton_repo_dir}/webhooks-extension/test"
+echo
+# Check that GOPATH and KO_DOCKER_REPO are set properly
+check GOPATH
+check KO_DOCKER_REPO "export KO_DOCKER_REPO=docker.io/${DOCKERHUB_USERNAME}"
```
Can you explain/comment/refactor this line please? Confusing on reading, could just be missing a newline

carlos-logro replied:
what's confusing here? here it is just checking that the user has correctly specified `GOPATH` and `KO_DOCKER_REPO`

a-roberts replied:
Should it not be

check KO_DOCKER_REPO
"export KO_DOCKER_REPO=docker.io/${DOCKERHUB_USERNAME}"

?

carlos-logro replied:
i see the confusion now. so the `check` method takes as arguments the value you want to check and optionally, an example of what that value looks like.

a-roberts commented on webhooks-extension/test/install_dashboard_and_extension.sh:
```diff
@@ -16,7 +16,36 @@ export test_dir="${tekton_repo_dir}/webhooks-extension/test"
+check GOPATH
+check KO_DOCKER_REPO "export KO_DOCKER_REPO=docker.io/${DOCKERHUB_USERNAME}"
+
+echo -e "\n\nInstalling the $VERSION_TEXT of the Dashoard and webhooks extension into $DASHBOARD_INSTALL_NS namespace.\n"
```
Tekton Dashboard*
Webhooks Extension

a-roberts commented on webhooks-extension/test/util.sh:
```diff
@@ -126,69 +126,103 @@ function install_tekton() {
+  version=$2
+  # Current Release
+  if [[ $version == "1" ]]; then
+    kubectl apply --filename https://github.com/tektoncd/dashboard/releases/download/v0.2.0/release.yaml
```
Is there a *latest release* URL we could use instead so we needn't keep updating this when we do cut a new one?

carlos-logro replied:
going to test with: `https://github.com/tektoncd/dashboard/releases/latest`

a-roberts commented on webhooks-extension/test/util.sh:
```diff
@@ -126,69 +126,103 @@ function install_tekton() {
+    # Check if user has dashboard source cloned
+    dashboard_dir=$GOPATH/src/github.com/tektoncd/dashboard
+    if [ -d $dashboard_dir ]; then
+      # Notify and depending on [version] we will ko apply or kubectl apply gcr yaml
```
Notify? Notify what

carlos-logro replied:
notifiy the user that they in fact have the dashboard cloned.

a-roberts commented on webhooks-extension/test/util.sh:
```diff
@@ -126,69 +126,103 @@ function install_tekton() {
+    else
+      echo -e "\n\nDashboard source not detected in $dashboard_dir"
+      if [[ $version != "3" ]]; then
+        read -p "Do you wish to git clone and ko apply the master version? " -n 1 -r cloneAndApply
```
What does this line do? The flags after it, isn't immediately obvious and I'd rather not go read the manpage for read 😄

carlos-logro replied:
`-n` flag denotes how many characters to read from the user input
`-r` where the value of the input is going to be stored

a-roberts commented on webhooks-extension/test/util.sh:
```diff
@@ -126,69 +126,103 @@ function install_tekton() {
     docker login
-    npm install
+    npm ci
+    npm rebuild node-sass
```
Is this actually needed? Haven't done it in months

carlos-logro replied:
great point. will remove and see if everything works properly!

AlanGreene replied:
`npm rebuild` should only be needed (occasionally) when switching node versions

carlos-logro commented:
@a-roberts changes made!

a-roberts commented on webhooks-extension/test/install_dashboard_and_extension.sh:
```diff
@@ -16,7 +17,35 @@ export test_dir="${tekton_repo_dir}/webhooks-extension/test"
-echo "Installing Dashoard and webhooks extension into $DASHBOARD_INSTALL_NS namespace"
-install_webhooks_extension $DASHBOARD_INSTALL_NS
-install_dashboard $DASHBOARD_INSTALL_NS
+echo -e "\n\nPlease choose which versions you will like to install:"
```
you would* like to install

carlos-logro commented:
@a-roberts changes made and file moved to the `scripts` folder

a-roberts commented on webhooks-extension/README.md:
```diff
@@ -16,7 +16,10 @@ See our [Getting Started](https://github.com/tektoncd/experimental/blob/master/w
-[Installing Development Build (nightly)](./docs/InstallNightlyBuild.md)  
+[Installing Development Build (nightly)](./docs/InstallNightlyBuild.md)
+
+As a convenience, the **/test/install_dashboard_and_extension.sh** script can be
```
scripts/ not test

carlos-logro commented:
🤦‍♂️🤦‍♂️🤦‍♂️ on it

carlos-logro commented:
@a-roberts updated!

mnuttall commented:
Please rebase and we'll get this in asap. Thank you!

carlos-logro commented:
@mnuttall tested and all choices seem to work except 2 and, based on what Vince and Brandon told me, it seems to be a syntax error in the nightly build extension yaml. Other than that everything is working. Also, should we keep the script under `scripts`, I noticed that the script doesn't work if `test/install_prereqs` has not been ran yet.

carlos-logro commented:
Updated the script. Reason behind the error was because of an indentation error on the webhhooks extension nightly build yaml. I also moved the script to the test folder primary because for this script to run successfully the prerequisites should already be installed.
